src/DDTrace/Integrations/Redis/RedisIntegration.php

```diff
@@ -0,0 +1,835 @@
+            return RedisIntegration::traceCommand($this, 'xGroup', $args);
+        });
+        dd_trace('Redis', 'xInfo', function () {
+            $args = func_get_args();
+            return RedisIntegration::traceCommand($this, 'xInfo', $args);
+        });
+        dd_trace('Redis', 'xLen', function () {
+            $args = func_get_args();
+            return RedisIntegration::traceCommand($this, 'xLen', $args);
+        });
+        dd_trace('Redis', 'xPending', function () {
+            $args = func_get_args();
+            return RedisIntegration::traceCommand($this, 'xPending', $args);
+        });
+        dd_trace('Redis', 'xRange', function () {
+            $args = func_get_args();
+            return RedisIntegration::traceCommand($this, 'xRange', $args);
+        });
+        dd_trace('Redis', 'xRead', function () {
+            $args = func_get_args();
+            return RedisIntegration::traceCommand($this, 'xRead', $args);
+        });
+        dd_trace('Redis', 'xReadGroup', function () {
+            $args = func_get_args();
+            return RedisIntegration::traceCommand($this, 'xReadGroup', $args);
+        });
+        dd_trace('Redis', 'xRevRange', function () {
+            $args = func_get_args();
+            return RedisIntegration::traceCommand($this, 'xRevRange', $args);
+        });
+        dd_trace('Redis', 'xTrim', function () {
+            $args = func_get_args();
+            return RedisIntegration::traceCommand($this, 'xTrim', $args);
+        });
+        dd_trace('Redis', 'pSubscribe', function () {
+            $args = func_get_args();
+            return RedisIntegration::traceCommand($this, 'pSubscribe', $args);
+        });
+        dd_trace('Redis', 'publish', function () {
+            $args = func_get_args();
+            return RedisIntegration::traceCommand($this, 'publish', $args);
+        });
+        dd_trace('Redis', 'subscribe', function () {
+            $args = func_get_args();
+            return RedisIntegration::traceCommand($this, 'subscribe', $args);
+        });
+        dd_trace('Redis', 'pubSub', function () {
+            $args = func_get_args();
+            return RedisIntegration::traceCommand($this, 'pubSub', $args);
+        });
+
+        return Integration::LOADED;
+    }
+
+    public static function traceCommand($redis, $command, $args)
+    {
+        $scope = GlobalTracer::get()->startIntegrationScopeAndSpan(
+            RedisIntegration::getInstance(),
+            "Redis.$command"
+        );
+        $span = $scope->getSpan();
+        $span->setTag(Tag::SPAN_TYPE, Type::CACHE);
+        $span->setTag(Tag::SERVICE_NAME, 'redis');
+        $span->setTag('redis.command', $command);
+
+        self::setServerTags($span, $redis);
+
+        $span->setTag(Tag::RESOURCE_NAME, $command);
+
+        return TryCatchFinally::executePublicMethod($scope, $redis, $command, $args);
+    }
+
+    private static function setServerTags($span, $redis)
```

Comment:
Since this is only being used in one place above, I would suggest moving these two `setTag()` calls to line `823` and remove this method. :)